piwheels/master/the_scribe.py

```diff
@@ -159,16 +159,28 @@ def handle_index(self, queue):
         except IOError as e:
             self.logger.error(str(e))
         else:
-            if msg == 'PKGBOTH':
+            if msg.startswith('PKG'):
```

Comment:
Hmm, this feels ... too complex for a dispatch routine. To be honest, I was pushing it with the old version, but this feels too much. In my head, a dispatch routine like this should grab a message and then call an appropriate action but not engage in (too much) logic itself (it this, then that, otherwise blah; no, dispatchers should just be this message does that, that message does this, etc).

I can see why it's being done but I do feel like a lot of this logic should be "down there" away from the dispatcher. For instance, `self.db.delete_package` should simply be called from `self.delete_package`. The removal from `package_cache` ... could belong up here I suppose (given the addition to it below). Still, it does feel like too much logic here, and the general "this message: that action" structure is broken up by this. I'll have a go at revising this tonight.

Reply:
I did think that when I saw the diff :scream: